crates/networking/p2p/src/bootnodes.rs

```diff
@@ -55,6 +72,31 @@ impl Bootnodes {
             Bootnodes::Default => bootnodes,
             Bootnodes::None => vec![],
             Bootnodes::Custom(bootnodes) => bootnodes,
+            Bootnodes::Multiaddr(_) => vec![],
+        }
+    }
+
+    pub fn to_multiaddrs_lean(&self) -> Vec<Multiaddr> {
+        match self {
+            Bootnodes::Default => {
+                serde_yaml::from_str(include_str!("../resources/lean_peers.yaml"))
+                    .expect("should deserialize static lean peers")
+            }
+            Bootnodes::None => vec![],
+            Bootnodes::Custom(enrs) => {
+                let mut static_peers: Vec<Enr> =
+                    serde_yaml::from_str(include_str!("../resources/lean_peers.yaml"))
+                        .expect("should deserialize static lean peers");
+                static_peers.extend(enrs.iter().cloned());
+                to_multiaddrs(&static_peers)
+            }
+            Bootnodes::Multiaddr(multiaddrs) => {
+                let mut static_peers: Vec<Multiaddr> =
+                    serde_yaml::from_str(include_str!("../resources/lean_peers.yaml"))
+                        .expect("should deserialize static lean peers");
+                static_peers.extend(multiaddrs.clone());
+                static_peers
+            }
```

Comment:
```suggestion
    pub fn to_multiaddrs_lean(&self) -> Vec<Multiaddr> {
        match self {
            Bootnodes::Default => {
                serde_yaml::from_str(include_str!("../resources/lean_peers.yaml"))
                    .expect("should deserialize static lean peers")
            }
            Bootnodes::None => vec![],
            Bootnodes::Custom(enrs) => {
                to_multiaddrs(&enrs)
            }
            Bootnodes::Multiaddr(multiaddrs) => {
                multiaddrs
            }
```

if we are using custom peers don't we want to use the custom peers? As this seems like an intentional choice, so I think we should only pull the default peers file if the default option is selected